zip file md5  and get list files and check

dairoquintero opened the pull request.

Commit:
zip file md5  and get list files and check

Commit:
zip file md5  and get list files and check

Commit:
solve current directory

Commit:
solve current directory

Commit:
solve current directory

Commit:
solve current directory

dairoquintero commented on src/test/java/co/com/api/test/dto/GithubRepoListDto.java:
```diff
@@ -0,0 +1,28 @@
+  @JsonCreator
+  public GithubRepoListDto(@JsonProperty("name") String name,
+                           @JsonProperty("path") String path, @JsonProperty("sha") String sha) {
+    this.name = name;
```
Add this lines to desarialize the response correctly

Commit:
solve current directory

Commit:
solve current directory

Commit:
solve current directory

Commit:
solve current directory

dairoquintero commented on src/test/java/co/com/api/test/dto/GithubRepoListDto.java:
```diff
@@ -0,0 +1,28 @@
+  private String path;
+  private String sha;
+
+  @JsonCreator
```
Add this lines to desarialize the response correctly

Commit:
solve current directory

Commit:
solve current directory

Commit:
solve current directory

aperdomob commented on src/test/java/co/com/api/test/ApiGithubRepositoriesTest.java:
```diff
@@ -94,28 +104,92 @@ public void apiGetRepositoryFilterRepoTest() {
+      FileOutputStream os = new FileOutputStream(new File("." + fileLocation), false);
+      os.write(valueArray);
+      os.close();
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
```
why do you avoid the exception?

dairoquintero replied:
In cases when the file exists previously the file, and fail the creation, without this try-catch fail the pipeline after some tests

aperdomob replied:
why the file should exist?

dairoquintero replied:
locally , it is possible if file is download previously. in the pipeline y try without the try-cath and the test fail in this line for excepcion

aperdomob commented on src/test/java/co/com/api/test/ApiGithubRepositoriesTest.java:
```diff
@@ -94,28 +104,92 @@ public void apiGetRepositoryFilterRepoTest() {
+  public void apiGetRepositoryFilterListFilesRepoTest() {
+    String fileName = "LICENSE.txt";
+    String shaFile = "bb0f726067e4130d7c19ee9128ca3b44397e41d2";
+    GithubRepoListDto payload = GithubRepoListDto.builder()
```
payload is not a meanly name

dairoquintero replied:
Done

aperdomob commented on src/test/java/co/com/api/test/dto/GithubRepoListDto.java:
```diff
@@ -0,0 +1,28 @@
+import lombok.Builder;
+import lombok.Data;
+
+@Data
```
remove data

dairoquintero replied:
Done

aperdomob commented on src/test/java/co/com/api/test/dto/GithubRepoListDto.java:
```diff
@@ -0,0 +1,28 @@
+  private String sha;
+
+  @JsonCreator
+  public GithubRepoListDto(@JsonProperty("name") String name,
```
can you use Constructor annotation from lombok in order to avoid this boilerplate

dairoquintero replied:
Done add contrsuctor paramteters lombok

Commit:
add contructor lombok Dot

dairoquintero reviewed (commented):
Changes Done

Commit:
add contructor lombok Dot

Commit:
zip file md5 and get list files and check

Commit:
zip file md5 and get list files and check

Commit:
zip file md5 and get list files and check

aperdomob commented on src/test/java/co/com/api/test/dto/GithubRepoListDto.java:
```diff
@@ -0,0 +1,25 @@
+
+@Builder
+@EqualsAndHashCode
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
```
why do you need to create a private constructor?

dairoquintero replied:
I add these lines to avoid errors to deserialize from, reading about in a forum, and the recommendation is add this lines, and work correctly. without these lines an error occurs when deserializing

aperdomob replied:
but why private?
I think that should be public

dairoquintero replied:
change acceslevel to Public

dairoquintero reviewed (commented):
Done

dairoquintero committed:
# Please enter the commit message for your changes. Lines starting
# with '#' will be kept; you may remove them yourself if you want to.
# An empty message aborts the commit.
#
# On branch github_repositories
# Changes to be committed:
#	modified:   src/test/java/co/com/api/test/ApiGithubRepositoriesTest.java
#

dairoquintero committed:
delete line remove file

dairoquintero committed:
add debug Ci file

dairoquintero committed:
remove filters reports

dairoquintero committed:
add debug Ci file

dairoquintero committed:
delete line remove file

dairoquintero committed:
delete line remove file

dairoquintero committed:
delete line remove file

dairoquintero committed:
delete line remove file

dairoquintero committed:
change access level to public repoListDto

dairoquintero commented on src/test/java/co/com/api/test/dto/GithubRepoListDto.java:
```diff
@@ -0,0 +1,25 @@
+@Builder
+@EqualsAndHashCode
+@AllArgsConstructor(access = AccessLevel.PUBLIC)
+@NoArgsConstructor(access = AccessLevel.PUBLIC)
```
change acces level to public

dairoquintero reviewed (commented):
Done corrections
